### PR TITLE
fix: Do not show glossary within links [INGEST-1093]

### PR DESCRIPTION
**Pull request by untitaker**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ default_language_version:
   node: latest
 exclude: '\.snap|node_modules$'
 repos:
-  - repo: git://github.com/pre-commit/pre-commit-hooks
```
Comment (untitaker) [inline .pre-commit-config.yaml:5]: random attempt at fixing some pre-commit hooks, but they're still broken for me. I think the patch here is still absolutely necessary as `git://` is literally deprecated by github

**Comment by untitaker** [comment]
@imatwawana which page is that?

**Comment by untitaker** [comment]
please check again... unfortunately it seems you need to run `yarn clean` beforehand such that this takes effect.

File: src/gatsby/plugins/gatsby-remark-terms/index.js [hunk 2]
```diff
@@ -24,17 +24,72 @@ const PATTERN = Object.keys(TERMS)
...
+
+  let insideCustomLink = false;
+
   // If a text node is present in child nodes, check if an abbreviation is present
   for (let c = 0; c < node.children.length; c++) {
     const child = node.children[c];
+
+    // While iterating through children, keep track of whether we're in a
+    // custom link component or not. MDX components have their own AST type,
+    // which is indirectly documented my the remark-mdx plugin:
```
Comment (untitaker) [inline src/gatsby/plugins/gatsby-remark-terms/index.js:54]:
```suggestion
    // which is indirectly documented by the remark-mdx plugin:
```

**Review by imatwawana — APPROVED** [review]
Looks good!

**Pull request merged**